### PR TITLE
FDL: Add SwiftUI quickstart for link handling

### PR DESCRIPTION
**Pull request by andrewheard**
Added a SwiftUI-based quickstart for handling Firebase Dynamic Links.

### CODE REVIEW

File: dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift [hunk 1]
```diff
@@ -16,27 +16,58 @@ import SwiftUI
...
+  func handleLink(_ url: URL) async -> DynamicLink? {
+    await withCheckedContinuation { continuation in
+      handleLink(url) { dynamicLink, error in
+        // TODO: Handle error
+        continuation.resume(returning: dynamicLink)
+      }
+    }
+  }
+
+  func handleLink(_ url: URL, completion: @escaping (DynamicLink?, Error?) -> Void) {
```
Comment (paulb777) [inline dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift:65]: Can this be `async throws` instead?

**Review by paulb777 — COMMENTED** [review]
Mostly looks good to me. 

Any updates needed for the instructions at https://github.com/firebase/quickstart-ios/blob/master/dynamiclinks/README.md beyond the iOs and Swift versions?

File: dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift [hunk 2]
```diff
@@ -43,8 +47,17 @@ struct DynamicLinksExampleApp: App {
...
-                receivedLinkModel = ReceivedLinkModel(receivedURL: url, dynamicLink: dynamicLink)
+              do {
+                if let dynamicLink: DynamicLink = try await handleLink(url) {
+                  receivedLinkModel = ReceivedLinkModel(
+                    receivedURL: url,
+                    dynamicLink: dynamicLink
+                  )
+                }
+              } catch UniversalURLError.notHandledByFDL {
+                print("FIRDynamicLinks is not handling the link.")
```
Comment (paulb777) [inline dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift:58]: Don't use Objective-C "FIR" prefix here and below.

Comment (andrewheard) [reply dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift]: This code no longer exists but noted for the future.

File: dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift (hunk 1 above)
Comment (andrewheard) [reply dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift]: Good idea. This code has moved to [`extractDynamicLink`](https://github.com/firebase/quickstart-ios/pull/1428/files#diff-e25e2955aedb4999ed4a8c5d7f7bc26a9b474bfaaf1a0f92dc768e02ecc00100R101) but I made the change there.

**Comment by andrewheard** [comment]
> Mostly looks good to me.
> 
> Any updates needed for the instructions at https://github.com/firebase/quickstart-ios/blob/master/dynamiclinks/README.md beyond the iOs and Swift versions?

I updated the iOS version info and added a note about setting the `domainURIPrefix` in the SwiftUI example (same as the `DOMAIN_URI_PREFIX` constant in the previous ones).

**Review by morganchen12 — APPROVED** [review]
LGTM, please let me know if the snippets you added need to be in docs somewhere.

**Comment by andrewheard** [comment]
> LGTM, please let me know if the snippets you added need to be in docs somewhere.

Thanks, Morgan. No need to update the snippets quite yet (just added them to prepare for FDL + SwiftUI setup docs in the future).

**Pull request merged**